src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -37,16 +37,36 @@ public class PersonCard extends UiPart<Region> {
     @FXML
     private FlowPane relationship;
 
+    // Callback to notify when a person is clicked
+    private final PersonSelectionHandler personSelectionHandler;
+
     /**
      * Creates a {@code PersonCode} with the given {@code Person} and index to display.
+     * It also allows handling of selection events through the provided {@code PersonSelectionHandler}.
      */
-    public PersonCard(Person person, int displayedIndex) {
+    public PersonCard(Person person, int displayedIndex, PersonSelectionHandler personSelectionHandler) {
         super(FXML);
         this.person = person;
+        this.personSelectionHandler = personSelectionHandler;
+
         id.setText(displayedIndex + ". ");
         name.setText(person.getName().fullName);
         phone.setText(person.getPhone().value);
         email.setText(person.getEmail().value);
         relationship.getChildren().add(new Label(person.getRelationship().relationship));
+
+        // Handle click events on the person card
+        getRoot().setOnMouseClicked(event -> {
+            // Notify the personSelectionHandler when a card is clicked
+            personSelectionHandler.handlePersonSelection(person);
+        });
+    }
+
+    /**
+     * Functional interface for person selection handler
+     */
+    @FunctionalInterface
```

Comment:
I see that both `PersonListPanel` and `PersonCard` are supposed to have a selection handler, but where is the implementation of the handler?

Reply:
The handler is basically `displayPersonDetail` in `MainWindow`.
Cause we want to send the `Person` data to `PersonDetailView` and display it on the `MainWindow`. The `PersonCard` and `PersonListPanel` is instantiated with the `displayPersonDetail` function

Reply:
I agree that the method should be renamed better to avoid confusion.